src/Migrations/M032MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,360 @@
+
+                $blocks = $parser->parse($post->post_content);
+
+                if (!is_array($blocks)) {
+                    throw new \Exception("Invalid block structure for post #" . $post->ID);
+                }
+
+                foreach ($blocks as &$block) {
+                    // Check if the block is valid.
+                    if (!is_array($block)) {
+                        continue;
+                    }
+
+                    // Check if the block has a 'blockName' key.
+                    if (!isset($block['blockName'])) {
+                        continue;
+                    }
+
+                    // Check if the block is a Social Media block.
+                    if ($block['blockName'] !== Utils\Constants::BLOCK_SOCIAL_MEDIA) {
+                        continue;
+                    }
+
+                    // Check if the embed is a Facebook page, in which case we do nothing.
+                    if (isset($block['attrs']['embed_type']) && $block['attrs']['embed_type'] === 'facebook_page') {
+                        continue;
+                    }
+
+                    // If not, we get the social media URL.
+                    $social_media_url = self::get_social_media_url($block);
+
+                    // Identify the source of the media (Facebook, Instagram, Twitter, X)
+                    $source = self::identify_source($social_media_url);
+
+                    // If there is no source or social media url, the block is transformed into an empty Embed block.
+                    if (!$source || !$social_media_url) {
+                        $block = self::transform_block_to_empty_embed($block);
+                        continue;
+                    }
+
+                    // If the source is Facebook or Instagram we do nothing.
+                    // We only want to migrate Twitter/X embeds.
+                    if ($source === self::SOURCE_FACEBOOK || $source === self::SOURCE_INSTAGRAM) {
+                        continue;
+                    }
+
+                    // If the source is X, we want to change it to Twitter.
+                    // Links from x.com don't work in the Embed block at the moment.
+                    if ($source === self::SOURCE_X) {
+                        $social_media_url = str_replace(
+                            self::SOURCE_X . '.com',
+                            self::SOURCE_TWITTER . '.com',
+                            $social_media_url,
+                        );
+                    }
+
+                    // Transform the Social Media block into an Embed block.
+                    $block = self::transform_blocks($block, $social_media_url);
+                }
+
+                // Unset the reference to prevent potential issues.
+                unset($block);
+
+                // Serialize the blocks content.
+                $new_content = serialize_blocks($blocks);
+
+                $post_update = array(
```

Comment:
Let's add an extra condition here to avoid unnecessary operations:

```php
if ($post->post_content === $new_content) {
         continue;
}
```